### PR TITLE
Adds a Filtered property and visual indicator to PropretyColumn in FluentDataGrid

### PR DESCRIPTION
**Pull request by NickHirras**
# Pull Request

## 📖 Description

Adds a Filtered boolean property to the PropertyColumn component of the FluentDataGrid.  When set to true, it renders a visual indicator on the column header filter button, letting the user know this column is currently filtered.

The Demo for _DataGrid Typical Usage_ has been updated to demonstrate this paramter.

See RFC

https://github.com/microsoft/fluentui-blazor/issues/1620

https://github.com/microsoft/fluentui-blazor/assets/3911628/81b16ad8-1255-4def-9885-2e092fe1dbd5

### 🎫 Issues

https://github.com/microsoft/fluentui-blazor/issues/1620

### General

- [ ] I have added tests for my changes.
- [X] I have tested my changes.
- [X] I have updated the project documentation to reflect my changes.
- [X] I have read the [CONTRIBUTING](https://github.com/Microsoft/fluentui-blazor/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have added [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for my new compontent
- [X] I have modified an existing component
- [X] I have validate [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for an existing component

### CODE REVIEW

**Comment by NickHirras** [comment]
Added same comment to the RFC. Does it make sense to use this red dot as the visual indicator, or should a new badge be created?

File: src/Core/Components/DataGrid/Columns/ColumnBase.razor [hunk 1]
```diff
@@ -20,7 +20,16 @@
             @if (ColumnOptions is not null && (Align == Align.Start || Align == Align.Center))
             {
                 <FluentButton Appearance="Appearance.Stealth" class="col-options-button" @onclick="@(() => Grid.ShowColumnOptionsAsync(this))" aria-label="Filter this column">
-                    <FluentIcon Value="@(new CoreIcons.Regular.Size24.Filter())" />
+                    @if(Filtered.GetValueOrDefault())
```
Comment (dvoituron) [inline src/Core/Components/DataGrid/Columns/ColumnBase.razor:23]: I find this way of writing clearer for testing the same thing.  What do you think?

```csharp
@if(Filtered == true)
```

**Review by dvoituron — CHANGES_REQUESTED** [review]
I also think it's nicer to use another icon instead of a badge. For example, `FilterAdd` can be used to explain that this column has added a filter.

image

**Comment by vnbaaij** [comment]
@NickHirras I agree with Denis. Use the available Filter icons. There are sort of a badge as well but more clear then a red dot.
...But I would say to use `FilterDismiss` for when a filter is set

File: src/Core/Components/DataGrid/Columns/ColumnBase.razor (hunk 1 above)
Comment (NickHirras) [reply src/Core/Components/DataGrid/Columns/ColumnBase.razor:23]: I agee will change that

**Comment by NickHirras** [comment]
@vnbaaij when trying to reference the FilterDismiss icon, it doesn't exist and getting a compiler error. Is this part of the Project Trimming / Generators?  What do I need to do to make the FilterDismiss icon available in CoreIcons?

**Comment by dvoituron** [comment]
> When trying to reference the FilterDismiss icon, it doesn't exist and getting a compiler error. Is this part of the Project Trimming / Generators? What do I need to do to make the FilterDismiss icon available in CoreIcons?

In fact, all "internal" icons are located in the file `CoreIcons.cs`.
In the beginning we used the **AssetsGenerator** project to create this file. 
But now it only contains occasional updates.
So, you can manually put your code into this file, and you can remove the comment `<auto-generated>...`

**Comment by NickHirras** [comment]
Here's latest version, using the `Filter` and `FilterDismiss` icons. 

No filter applied:

image

With a filter applied:

image

**Review by dvoituron — APPROVED** [review]

**Pull request merged**